pkg/cloud/cloud.go

```diff
@@ -1621,25 +1677,53 @@ func (c *cloud) waitForVolumeModification(ctx context.Context, volumeID string)
 	return nil
 }
 
+func describeVolumesModifications(ctx context.Context, svc EC2API, request *ec2.DescribeVolumesModificationsInput) ([]types.VolumeModification, error) {
+	volumeModifications := []types.VolumeModification{}
+	var nextToken *string
+	for {
+		response, err := svc.DescribeVolumesModifications(ctx, request)
+		if err != nil {
+			if isAWSErrorModificationNotFound(err) {
```

Comment:
Is it safe to do this here? What happens if we batch volumes `A` and `B` and only one is being modified, will we return that both are not being modified?

Reply:
From my understanding, it is safe because for any volume V: 
A. If a DVM call containing V is batched, either
i. There has been atleast 1 volume modification on V before, and therefore DVM would never return `ErrorModificationNotFound`
ii. ModifyVolume would have been called and synchronously succeeded for V, which we know means DVM would return a volume modification with either Pending, Optimizing, Completed, or Failed states (Because ModifyVolume and DVM are consistent with eachother). Never `ErrorModificationNotFound` 

B. Therefore, this block of code only gets hit by pre-modify DVM calls which are never batched. 

---

There are two things we can do to be extra safe if you spot a flaw in my reasoning above:
1. Pass in that `isBatched`/`batchable` boolean into this function and only return `VolumeNotBeingModified` if this is a non-batchable call (if it is a batchable call, return another error such that we don't pre-emptively return true [here](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/c424cbfe5d9e9d7693b104f9b33857c502af08b7/pkg/cloud/cloud.go#L1603)).
2. Have two separate `describeVolumesModifications` (one for the pre-modification DVM call, one for the post-modify DVM call)

I prefer option 1 to option 2, but I'm not convinced either are necessary.